File: deamhome/src/main/java/sosteam/deamhome/global/entity/BaseEntity.java
package sosteam.deamhome.global.entity;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

public abstract class BaseEntity {
    @Id
    private String id;

    @CreatedDate
    private final LocalDateTime createdAt = LocalDateTime.now();

    @LastModifiedDate
    private final LocalDateTime updatedAt = LocalDateTime.now();
}

File: src/main/kotlin/sosteam/deamhome/global/entity/BaseEntity.kt
package sosteam.deamhome.global.entity

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.relational.core.mapping.Column
import java.time.LocalDateTime

abstract class BaseEntity : Domain {
	@CreatedDate
	@Column("created_at")
	private var createdAt = LocalDateTime.now()
	
	@LastModifiedDate
	@Column("updated_at")
	private var updatedAt = LocalDateTime.now()
	
	fun getCreatedAt(): LocalDateTime {
		return createdAt
	}
}